pkg/k8shandler/kibana/secret.go

```diff
@@ -27,35 +27,45 @@ var secretCertificates = map[string]map[string]string{
 func (clusterRequest *KibanaRequest) GetSecret(secretName string) (*core.Secret, error) {
 	secret := &core.Secret{}
 	if err := clusterRequest.Get(secretName, secret); err != nil {
-		if errors.IsNotFound(err) {
+		if apierrors.IsNotFound(kverrors.Root(err)) {
 			return nil, err
 		}
-		return nil, fmt.Errorf("Failed to get %v secret: %v", secret.Name, err)
+		return nil, kverrors.Wrap(err, "failed to get secret",
+			"name", secret.Name,
+		)
 	}
 
 	return secret, nil
 }
 
-func (clusterRequest *KibanaRequest) readSecrets() (err error) {
+// readSecrets reads all of the secrets it can find within secretCertificates
+// if any of the secrets are not found then they are ignored
+func (clusterRequest *KibanaRequest) readSecrets() error {
 
 	for secretName, certMap := range secretCertificates {
-		if err = clusterRequest.extractCertificates(secretName, certMap); err != nil {
-			return
+		err := clusterRequest.extractCertificates(secretName, certMap)
+		if err == nil {
+			continue
+		}
```

Comment:
is this necessary to do since we don't have any other logic if the err != nil (past seeing if its a not found error)
`if err != nil && !apierrors.IsNotFound ... `